webservices/resources/legal.py

```diff
@@ -74,42 +75,50 @@ def get(self, q, from_hit=0, hits_returned=20, type='all', **kwargs):
         parsed_query = parse_query_string(q)
         terms = parsed_query.get('terms')
         phrases = parsed_query.get('phrases')
+        hits_returned = min([200, hits_returned])
 
         results = {}
         total_count = 0
         for type in types:
-            query = {"query": {"bool": {
-                     "must": [
-                         {"term": {"_type": type}},
-                         ],
-                     "should": [
-                         {"match": {"no": q}},
-                         {"match_phrase": {"_all": {"query": q, "slop": 50}}},
-                         ]
-                     }},
-                "highlight": {"fields": [{"text": {}},
-                    {"name": {}}, {"number": {}}]},
-                "_source": {"exclude": "text"}}
+            must_query = [Q('term', _type=type)]
+            text_highlight_query = Q()
 
             if len(terms):
-                query['query']['bool']['must'].append({"match": {"_all": ' '.join(terms)}})
-
-            for phrase in phrases:
-                query['query']['bool']['must'].append({"match_phrase": {"text": phrase}})
-
-            hits_returned = min([200, hits_returned])
-            es_results = es.search(query, index='docs', size=hits_returned,
-                             es_from=from_hit)
-            hits = es_results['hits']['hits']
-            for hit in hits:
-                highlights = []
-                if 'highlight' in hit:
-                    for key in hit['highlight']:
-                        highlights.extend(hit['highlight'][key])
-                hit['_source']['highlights'] = highlights
-            count = es_results['hits']['total']
+                term_query = Q('match', _all=' '.join(terms))
+                must_query.append(term_query)
+                text_highlight_query = text_highlight_query & term_query
```

Comment:
It's just some syntax sugar from the DSL http://elasticsearch-dsl.readthedocs.io/en/latest/search_dsl.html#query-combination